### PR TITLE
JAMES-3878 - Remove icu4j dependency

### PR DESCRIPTION
**Pull request by woj-tek**
Remove icu4j dependency and replace required functionality with other dependencies

https://issues.apache.org/jira/projects/JAMES/issues/JAMES-3878

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotationKey.java [hunk 1]
```diff
@@ -25,21 +25,18 @@
...
 
 public class MailboxAnnotationKey {
 
     public static final String SLASH_CHARACTER = "/";
     public static final String TWO_SLASH_CHARACTER = "//";
 
-    private static final UnicodeSet NAME_ANNOTATION_PATTERN = new UnicodeSet("[[a-z][A-Z][0-9]]")
-            .add(SLASH_CHARACTER)
-            .freeze();
+    private static final Pattern NAME_ANNOTATION_PATTERN = Pattern.compile("^([\\w\\d/]*)$");
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotationKey.java:35]: My preference goes to relying on Guava CharMatcher and avoid REGEX alltogether if that works.

Comment (chibenwa) [reply mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotationKey.java:35]: For instance:

```
    private static final CharMatcher NAME_ANNOTATION_PATTERN = CharMatcher.inRange('a', 'z')
        .or(CharMatcher.inRange('A', 'Z'))
        .or(CharMatcher.inRange('0', '9'))
        .or(CharMatcher.is('/'));
```

Passes the unit tests

**Review by Arsnael — COMMENTED** [review]
Read it, nothing to add :)

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotationKey.java (hunk 1 above)
Comment (woj-tek) [reply mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotationKey.java]: Switched to CharMatcher

**Review by chibenwa — APPROVED** [review]

**Pull request merged**